components/chat_ui/browser/chat_ui_api_request.cc

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/chat_ui/browser/chat_ui_api_request.h"
+
+#include <base/containers/flat_map.h>
+
+#include <utility>
+
+#include "base/debug/dump_without_crashing.h"
+#include "base/json/json_writer.h"
+#include "brave/components/chat_ui/browser/buildflags.h"
+#include "brave/components/chat_ui/common/chat_ui_constants.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("chat_ui", R"(
+      semantics {
+        sender: "ChatUI"
+        description:
+          "This is used to communicate with our partner API"
+          "on behalf of the user interacting with the ChatUI."
+        trigger:
+          "Triggered by user sending a prompt."
+        data:
+          "Will generate a text completion that attempts to match whatever context or pattern the user gave it"
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+GURL GetURLWithPath(const std::string& host, const std::string& path) {
+  return GURL(std::string(url::kHttpsScheme) + "://" + host).Resolve(path);
+}
+
+std::string CreateJSONRequestBody(base::ValueView node) {
+  std::string json;
+  base::JSONWriter::Write(node, &json);
+  return json;
+}
+
+}  // namespace
+
+namespace chat_ui {
+
+ChatUIAPIRequest::ChatUIAPIRequest(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : api_request_helper_(GetNetworkTrafficAnnotationTag(),
+                          url_loader_factory) {}
+
+ChatUIAPIRequest::~ChatUIAPIRequest() = default;
+
+void ChatUIAPIRequest::QueryPrompt(ResponseCallback callback,
+                                   const std::string& prompt) {
+  auto internal_callback =
+      base::BindOnce(&ChatUIAPIRequest::OnGetResponse,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  base::Value::Dict dict;
+  base::Value::List stop_sequences;
+  stop_sequences.Append("Human:");
+
+  dict.Set("prompt", prompt);
+  dict.Set("max_tokens_to_sample", 200);
+  dict.Set("temperature", 1);
```

Comment:
we might want to go with a lower temperature given that the task we want to focus on (summarisation) is more analytical than creative/generative. For summarization, OpenAI models usually default to 0.7 on a [0,2] range

Reply:
done